### ISSUE TITLE
Not functional for BD on Windows

### ISSUE THREAD

**Issue by Yentis** [binarynoise/open-in-mpv#11]
Some issues I encountered trying this plugin:
- doesn't work for embeds, only plain urls
- setup.sh doesn't download a file, just opens a page and user needs to right click -> save as
- windows machines can't run sh files by default (needs something like git installed)
- the context menu button doesn't work, it opens a command prompt that just throws an error
![mpv_3Cpn5ClPsS](https://github.com/user-attachments/assets/1e97be82-121a-46f3-a2f1-440e530342b7)
![mpv_ysBpAOb9Es](https://github.com/user-attachments/assets/bec902aa-d1a5-420c-83df-4fcb807f1898)

**Comment by Joshix-1** [comment]
> doesn't work for embeds, only plain urls

Good idea, that would be a new feature

> setup.sh doesn't download a file, just opens a page and user needs to right click -> save as

I'm not aware of a solution to this problem. Browser extensions cannot install files locally.

> windows machines can't run sh files by default (needs something like git installed)

Windows is not supported.

> the context menu button doesn't work, it opens a command prompt that just throws an error

Windows is not supported.

**Comment by Yentis** [comment]
This [plugin](https://github.com/binarynoise/open-in-mpv/blob/main/open-in-mpv.plugin.js) was submitted to BetterDiscord (https://betterdiscord.app/plugin/open%20in%20mpv), discord's userbase is predominantly Windows so it should be supported or at least mentioned very clearly that it's not supported.
> I'm not aware of a solution to this problem. Browser extensions cannot install files locally.

A BD plugin can use `fs.writeFile` (https://docs.betterdiscord.app/plugins/introduction/environment#node-js)

**Comment by binarynoise** [comment]
I put it onto the post-publishing TODO.

As we don't use Windows, adding this feature is not a priority, however it seems doable.
A first start would be https://stackoverflow.com/questions/80650/how-do-i-register-a-custom-url-protocol-in-windows

**Comment by binarynoise** [comment]
> the context menu button doesn't work, it opens a command prompt that just throws an error

Did you, @Yentis, at some point instruct windows to use `cmd` to handle the `mpv://` url?

**Comment by Yentis** [comment]
Yes, I had a protocol registered in the past. I removed the remnants of it and now it shows this instead:
![explorer_xqOn9Mwrz0](https://github.com/user-attachments/assets/9044b921-8a38-4933-bc84-edfc45d18ee0)
No app can be selected.

**Comment by binarynoise** [comment]
good, I'll experiment with windows when I find some time.
